File: server/app/src/main/java/com/borkdominik/big/glsp/uml/core/ImportModelAction.java
/********************************************************************************
 * Copyright (c) 2024 borkdominik and others.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v. 2.0 which is available at
 * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
 * available at https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: EPL-2.0 OR MIT
 ********************************************************************************/
package com.borkdominik.big.glsp.uml.core;

import com.borkdominik.big.glsp.server.core.handler.action.new_file.BGNewFileResponseAction;
import org.eclipse.glsp.server.actions.RequestAction;

import java.util.HashMap;
import java.util.Map;

public class ImportModelAction extends RequestAction<BGNewFileResponseAction> {
   public static final String KIND = "importModel";

   private final Map<String, String> options;

   public ImportModelAction() {
      super(KIND);
      options = new HashMap<>();
   }

   public ImportModelAction(final Map<String, String> options) {
      super(KIND);
      this.options = options;
   }

   public Map<String, String> getOptions() { return options; }
}

File: client/packages/uml-protocol/src/action-protocol/import-model.action.ts
/*********************************************************************************
 * Copyright (c) 2023 borkdominik and others.
 *
 * This program and the accompanying materials are made available under the
 * terms of the MIT License which is available at https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 *********************************************************************************/
import { Action, JsonMap, RequestAction, ResponseAction } from '@eclipse-glsp/protocol';

export interface ImportModelAction extends RequestAction<ImportModelResponseAction> {
    kind: typeof ImportModelAction.KIND;
    options?: JsonMap;
}
export namespace ImportModelAction {
    export const KIND = 'importModel';

    export function importModel(jsonModelUri: string, filename: string, modelUri: string): ImportModelAction {
        return {
            kind: KIND,
            options: {
                jsonModelUri: jsonModelUri,
                filename: filename,
                modelUri: modelUri
            },
            requestId: ''
        };
    }
}

export interface ImportModelResponseAction extends ResponseAction {
    kind: typeof ImportModelResponseAction.KIND;
    sourceUri: string;
}

export namespace ImportModelResponseAction {
    export const KIND = 'importModelResponse';

    export function is(action: unknown): action is ImportModelResponseAction {
        return Action.hasKind(action, KIND);
    }
}
